felles/src/main/kotlin/no/nav/familie/kontrakter/felles/Periode.kt

```diff
@@ -2,113 +2,158 @@ package no.nav.familie.kontrakter.felles
 
 import java.time.LocalDate
 import java.time.YearMonth
+import java.time.temporal.Temporal
 
-data class Periode(
-    val fomDato: LocalDate,
-    val tomDato: LocalDate
-) : Comparable<Periode> {
+sealed class Periode<T>(
+    open val fom: T,
+    open val tom: T,
+) : Comparable<Periode<T>> where T : Comparable<T>, T : Temporal {
 
-    val fomMåned = YearMonth.from(fomDato)
-    val tomMåned = YearMonth.from(tomDato)
+    val fomMåned get() = YearMonth.from(fom)
+    val tomMåned get() = YearMonth.from(tom)
 
-    constructor(fomMåned: YearMonth, tomMåned: YearMonth) : this(fomMåned.atDay(1), tomMåned.atEndOfMonth())
-
-    constructor(måned: YearMonth) : this(måned.atDay(1), måned.atEndOfMonth())
-
-    init {
-        require(tomDato >= fomDato) { "Til-og-med før fra-og-med: $fomDato > $tomDato" }
-    }
-
-    infix fun inneholder(dato: LocalDate): Boolean {
-        return dato in fomDato..tomDato
+    protected fun validate() {
+        require(tom >= fom) { "Til-og-med før fra-og-med: $fom > $tom" }
```

Comment:
Fint med validering, bare husk at perioder fra infotrygd ikke nødvendigvis har tom >= fom

Reply:
Jeg så det, så denne kan nok ikke brukes for innfotrygd-perioder.